chain/chain/src/chain.rs

```diff
@@ -573,7 +573,7 @@ impl Chain {
         let resharding_manager =
             ReshardingManager::new(chain_store.store(), epoch_manager.clone(), resharding_sender);
 
-        let num_shards = shard_layout.num_shards() as usize;
+        let num_shards = runtime_adapter.get_shard_layout(PROTOCOL_VERSION).num_shards() as usize;
```

Comment:
nit: This is a tiny bit risky, someone may use this as the current number of shards. Can you rename to something more telling? Like binary_max_num_shards or just max_num_shards? And add a comment please.